src/lib.rs

```diff
@@ -490,29 +501,31 @@ pub trait PwmAccess<T> {
 impl<DV: DeviceVariant, I: RegisterAccess> PwmAccess<u16> for Lp586x<DV, I, DataMode16Bit> {
     type Error = I::Error;
 
-    fn set_pwm(&mut self, start: u16, values: &[u16]) -> Result<(), Self::Error> {
-        let mut registers = [0; 396];
+    fn set_pwm(&mut self, start_dot: u16, values: &[u16]) -> Result<(), Self::Error> {
+        let mut buffer = [0; Variant0::NUM_DOTS as usize * 2];
 
-        if values.len() + start as usize > (DV::NUM_DOTS as usize) {
+        if values.len() + start_dot as usize > (DV::NUM_DOTS as usize) {
+            // TODO: probably we don't want to panic in an embedded system...
             panic!("Too many values supplied for given start and device variant.");
```

Comment:
I think we could just ignore any excess data here (or return an Error), but in either way this behaviour should be documented.

Reply:
Yep.. but the error type is generic (and unknown in this place i guess). Anothe way to go is just to silently ignore the excess data (truncate)

Reply:
Ah yes, I also noticed the return type is `Self::Error`, not `Error<>` after commenting. That does not seem right, I'll fix that in the future.